Objects/typeobject.c

```diff
@@ -6492,7 +6492,12 @@ slot_tp_hash(PyObject *self)
        Py_hash_t.  Therefore our transformation must preserve values that
        already lie within this range, to ensure that if x.__hash__() returns
        hash(y) then hash(x) == hash(y). */
-    h = PyLong_AsSsize_t(res);
+    if (Py_SIZE(res) < 0) {
+        h = PyLong_AsSsize_t(res);
+    }
+    else {
+        h = (Py_ssize_t)PyLong_AsSize_t(res);
```

Comment:
It looks like signed overflow can happen here.

Reply:
Yes, but it is everywhere in evaluating and using hashes.